src/ui/pages/Menu/components/ConnectWallet/ConnectWallet.tsx

```diff
@@ -125,9 +125,12 @@ const ConnectWallet = forwardRef<ConnectWalletOptionRef, object>(
       await Agent.agent.peerConnectionMetadataStorage.deletePeerConnectionMetadataRecord(
         data.id
       );
-      const connections =
-        await Agent.agent.peerConnectionMetadataStorage.getAllPeerConnectionMetadata();
-      dispatch(setWalletConnectionsCache(connections));
+
+      dispatch(
+        setWalletConnectionsCache(
+          connections.filter((connection) => connection.id !== data.id)
```

Comment:
Does this work? I thought with Redux we had to clone the object before updating Redux

Reply:
It works. The case we faced last time was different. `filter` creates a shallow copy of the array.

Reply:
good point